A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/EICAS_Common.js

```diff
@@ -79,16 +79,22 @@ class EICASCommonDisplay extends Airliners.EICASTemplateElement {
     refreshGrossWeight(_force = false) {
         let gw = 0;
         let isInMetric = BaseAirliners.unitIsMetric(Aircraft.A320_NEO);
+        let unit;
         if (isInMetric) {
-            gw = Math.round(SimVar.GetSimVarValue("TOTAL WEIGHT", "kg"));
-            if (this.gwUnit)
+            if (this.gwUnit) {
                 this.gwUnit.textContent = "KG";
-        }
-        else {
-            gw = Math.round(SimVar.GetSimVarValue("TOTAL WEIGHT", "lbs"));
-            if (this.gwUnit)
+                unit = "kg";
+            }
+        } else {
+            if (this.gwUnit) {
                 this.gwUnit.textContent = "LBS";
+                unit = "lbs";
+            }
+        }
+        for (var i = 1; i <= 6; i++) {
```

Comment:
Idk about leaving this hard coded. I assume these are all the different payload categories? While they shouldn't change, there could be a mod in the future that changes it which would break this. Is there a way to find this out from the sim?

Reply:
I'm looking on the sim variable table. Let's me try with the payload station number variable.